lib/thrift/parser.ex

```diff
@@ -41,26 +43,76 @@ defmodule Thrift.Parser do
   end
 
   @doc """
-  Parses a Thrift IDL file.
+  Parses a Thrift IDL file into its AST representation.
+  """
+  @spec parse_file(Path.t()) :: {:ok, Thrift.AST.Schema.t()} | error
+  def parse_file(path) do
+    case read_file(path) do
+      {:ok, contents} ->
+        parse_string(contents)
+
+      {:error, message} ->
+        {:error, {nil, message}}
+    end
+  end
 
-  In addition to the given file, all included files are also parsed and
-  returned as part of the resulting `Thrift.Parser.FileGroup`.
+  @doc """
+  Parses a Thrift IDL file and its included files into a file group.
   """
-  @spec parse_file(Path.t(), opts) :: FileGroup.t()
-  def parse_file(file_path, opts \\ []) do
+  @spec parse_file_group(Path.t(), opts) :: {:ok, FileGroup.t()} | error
+  def parse_file_group(path, opts \\ []) do
```

Comment:
Do we want to keep the module dependency parser -> file group now that we have a `parse_file` that returns AST? We require file group -> parser dependency to handle includes so have an existing circular dependency. Its quite awkward as `FileGroup` takes AST as input but then does its own parsing on files for includes. For a cleaner API I think we should always use only one type of input. As we need to parse included files in `FileGroup` the input would be paths. Therefore I think that we should resolve this by moving this function to `FileGroup` and only supporting paths as input to `FileGroup` functions, and `FileGroup` would then call `Parser`.

This should also make it cleaner to support multiple root schemas for a single file group with common files that we discussed.

Reply:
Yes, I started thinking about this a little in context #236. I don't have any strong feelings yet, but I agree it's going to take a bit of thought.